include/envoy/upstream/outlier_detection.h

```diff
@@ -81,8 +93,14 @@ class DetectorHostMonitor {
    * @return the success rate of the host in the last calculated interval, in the range 0-100.
    *         -1 means that the host did not have enough request volume to calculate success rate
    *         or the cluster did not have enough hosts to run through success rate outlier ejection.
+   * @param type specifies for which Success Rate Monitor the success rate value should be returned.
+   If the outlier detector is configured not to split external and local origin errors,
+   ExternalOrigin type returns success rate for all types of errors: external and local origin and
+   LocalOrigin type returns -1. If the outlier detector is configured to split external and local
```

Comment:
Instead of returning -1 here, would it be better to use absl::optional<>? Maybe something to think about as a follow up.

Reply:
That -1 is displayed on GUI screen by admin.cc. If nobody interprets it, I can change it (would prefer another PR).